api/server.js

```diff
@@ -1,3 +1,92 @@
 // BUILD YOUR SERVER HERE
+const express = require('express')
+const Users = require(`./users/model`)
+const server = express()
 
-module.exports = {}; // EXPORT YOUR SERVER instead of {}
+server.use(express.json())
+
+//Endpoints:
+// GET requests:
+server.get(`/api/users`, async (req, res) => {
+   try {
+      const users = await Users.find()
+      res.status(200).json(users)
+   } catch (err) {
+      res.status(500).json({ message: "The users information could not be retrieved" })
+   }
+})
+
+server.get(`/api/users/:id`, async (req, res) => {
+   try {
+      const { id } = req.params
+      const userId = await Users.findById(id)
+      if (!userId) {
+         res.status(404).json({ message: "The user with the specified ID does not exist" })
+      } else {
+         res.status(200).json(userId)
+      }
+   }
+   catch (err) {
+      res.status(500).json({ message: "The user information could not be retrieved" })
+   }
+})
+
+
+//POST Request:
+server.post(`/api/users`, async (req, res) => {
+   try {
+      const { name, bio } = req.body
+      const createdUser = await Users.insert({ name, bio })
+
+      if (!name || !bio) {
+         res.status(400).json({ message: "Please provide name and bio for the user" })
+      }
+      else {
+         res.status(201).json(createdUser)
+      }
+   } catch (err) {
+      res.status(500).json({ message: "There was an error while saving the user to the database" })
+   }
+})
+
+//PUT Request:
+server.put(`/api/users/:id`, async (req, res) => {
+   try {
+      const { id } = req.params
+      const { name, bio } = req.body
+      const updatedUser = await Users.update(id, { name, bio })
+
+      if (!name || !bio) {
```

Comment:
Nit: same as the Post request about checking the name / bio before updating the db